internal/command/deploy/machines.go

```diff
@@ -539,18 +617,34 @@ func (md *machineDeployment) resolveUpdatedMachineConfig(origMachineRaw *api.Mac
 			Config: &api.MachineConfig{},
 		}
 	}
+	newConfig := machine.CloneConfig(origMachineRaw.Config)
 
 	launchInput := &api.LaunchMachineInput{
 		ID:      origMachineRaw.ID,
 		AppID:   md.app.Name,
 		OrgSlug: md.app.Organization.ID,
-		Config:  lo.Ternary(md.restartOnly, origMachineRaw.Config, &api.MachineConfig{}),
+		Config:  newConfig,
 		Region:  origMachineRaw.Region,
 	}
 
-	launchInput.Config.Metadata = md.defaultMachineMetadata()
-	for k, v := range origMachineRaw.Config.Metadata {
-		if !isFlyAppsPlatformMetadata(k) {
+	launchInput.Config.Statics = nil
+	if launchInput.Config.Metadata == nil {
+		launchInput.Config.Metadata = map[string]string{}
+	}
+
+	// Remove any platform metadata (such as release version)
+	// This way, when we're patching the config with defaults,
+	// we'll also end up overriding the platform metadata.
+	for k := range launchInput.Config.Metadata {
+		if isFlyAppsPlatformMetadata(k) {
+			delete(launchInput.Config.Metadata, k)
+		}
+	}
+	// Patch any holes in the config with defaults
+	defaultMetadata := md.defaultMachineMetadata()
+	for k, v := range defaultMetadata {
+		_, has := launchInput.Config.Metadata[k]
+		if !has {
 			launchInput.Config.Metadata[k] = v
 		}
 	}
```

Comment:
all this stuff can be rewritten as :

```go
launchInput.Config.Metadata = lo.Assign(
  lo.OmitBy(launchInput.Config.Metadata, func(k, v string) bool {
    return isFlyAppsPlatformMetadata(k)
  }),
  md.defaultMachineMetadata(),
) 
```

Reply:
oh that's sick, I haven't gotten a chance to use `lo` much but that's great